Yosemite/Yosemite/Tools/Products/Product+OrderItem.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+extension Product {
+    /// Converts a `Product` to an `OrderItem`
+    ///
+    public func toOrderItem(quantity: Decimal) -> OrderItem {
+        let price = NSDecimalNumber(string: price)
+        let total = quantity * price.decimalValue
+
+        return OrderItem(itemID: 0,
+                         name: name,
+                         productID: productID,
+                         variationID: 0,
+                         quantity: quantity,
+                         price: price,
+                         sku: nil,
+                         subtotal: "\(total)",
+                         subtotalTax: "",
+                         taxClass: "",
+                         taxes: [],
+                         total: "\(total)",
+                         totalTax: "0",
+                         attributes: []
+        )
+    }
```

Comment:
> Could we unit test this?

Good catch, I'll work on a test for this.

> Also, I wonder if we will have problems with the total if the product has a sale price or if later the user adds a tax rate 🤔

Hm, good thoughts! For the purposes of creating the order remotely it shouldn't matter, because we only send the ID and quantity to the API. I added the price and total amounts there for convenience when calculating the total on a new order. (The `price` property should be the current price, at least when I checked, so it reflects the sale price if the product is currently on sale.) But you're right that we should keep it in mind — we should definitely double check when we work on tax rates, to make sure it doesn't cause problems. 🤔